### ISSUE TITLE
[1.18.2] Affinity Value lookup causes client freezes on holding shift over items due to Immersive Engineering

### ISSUE THREAD

**Issue by DerCommander323** [daedalus4096/PrimalMagick#111]
While testing some stuff i noticed that on holding shift over most items it will cause the client to freeze for a few seconds (only once per item type), and after looking in the logs, it seems that this is due to Primal Magick trying to scan the items recipe for the Affinities, but IE does something that causes an IllegalArgumentException, which also freezes the game for a short time.

Log: https://mclo.gs/Y1M8alc
Versions:
Minecraft 1.18.2
Forge: 40.1.60
Primal Magick: 2.1.2
ImmersiveEngineering: 1.18.2-8.0.2-149

I also have LegendaryTooltips installed, in case it is related to that, but i dont think so

**Comment by EnderMelody** [comment]
i had this same issue and the immersive engineer devs pointing me to this mod. here is the [issue i opened for them](https://github.com/BluSunrize/ImmersiveEngineering/issues/5375#issuecomment-1193273787) .
along with my [error log](https://github.com/BluSunrize/ImmersiveEngineering/files/9175029/message.txt)

**Comment by FoxMcloud5655** [comment]
Can confirm this issue.  If more information is needed, I can provide it.

**Comment by daedalus4096** [comment]
This should be fixed in the upcoming 1.20.1 release.  Should you continue to encounter this sort of problem, please file a new ticket.  Thank you for your patience!
